lib/src/excel.dart

```diff
@@ -1,36 +1,139 @@
+        final tableBody = insertionOrderList.map(_generateTableRow).toList();
+
+        // Calculates table size and adds a master table.
+        final endAddressInt = _startAddressInt + _tableHeader.length - 1;
+        final endAddress =
+            '${String.fromCharCode(endAddressInt)}${tableBody.length + 1}';
+        final tableAddress = 'Query!$_startAddress:$endAddress';
+        final table = context.workbook.tables.add(tableAddress, true);
+
+        // Adds and formats table header.
+        table.getHeaderRowRange()
+          ..values = [_tableHeader]
+          ..format.font.name = _fontName
+          ..format.font.size = _fontSize
+          ..format.font.bold = true
+          ..format.horizontalAlignment = _horizontalAlignment
+          ..format.borders.getItem('EdgeTop').style = _borderStyle
+          ..format.borders.getItem('EdgeBottom').style = _borderStyle;
+
+        // Adds and formats table body.
+        table.getDataBodyRange()
+          ..formulas = tableBody
+          ..format.font.name = _fontName
+          ..format.font.size = _fontSize
+          ..format.horizontalAlignment = _horizontalAlignment;
+
+        // Auto-fits all used cells.
+        sheet.getUsedRange().getEntireColumn().format.autofitColumns();
+        sheet.getUsedRange().getEntireRow().format.autofitRows();
+
+        // Sets the sheet as active.
+        sheet.activate();
+        return context.sync();
+      }));
     }));
   }
 
-  Future<void> _populate(RequestContext context) async {
-    final sheet = context.workbook.worksheets.getActiveWorksheet();
-    final range = sheet.getRange('A1');
-    final values = <List<String>>[
-      ['test test']
-    ];
-    range.values = values;
-    range.format.autofitColumns();
-    return context.sync();
+  static List _generateTableRow(InsertionOrder io) => [
+        io.insertionOrderId,
+        io.advertiserId,
+        io.campaignId,
+        io.displayName,
+        io.entityStatus.toString(),
+        io.updateTime,
+        io.pacing.pacingPeriod.toString(),
+        io.pacing.pacingType.toString(),
+        _calculatePacingDailyMax(io.pacing.dailyMaxMicros),
+        io.pacing.dailyMaxImpressions,
+        io.budget.budgetUnit.toString(),
+        io.budget.automationType.toString(),
+        _calculateTotalBudgetAmount(io.budget.budgetSegments),
+        _calculateStartDate(io.budget.budgetSegments.first),
+        _calculateEndDate(io.budget.budgetSegments.last),
+      ];
+
+  static String _calculatePacingDailyMax(String dailyMaxMicros) =>
+      dailyMaxMicros.isEmpty
+          ? dailyMaxMicros
+          : (Int64.parseInt(dailyMaxMicros) * 1e-6).toString();
+
+  static String _calculateTotalBudgetAmount(
+      List<InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment>
+          budgetSegments) {
+    var budgetAmountList = budgetSegments
+        .map((segment) => Int64.parseInt(segment.budgetAmountMicros))
+        .toList();
+    var totalBudgetMicros =
```

Comment:
same

Reply:
can you merge these list comps?
reduce (value + Int64(parseInt(element.budgetAmountMicros))

Reply:
Reducer needs to return the same type as the list, but now the map & reduce are chained together so we don't the intermediate variable.